modules/rhythmoneBidAdapter.js

```diff
@@ -1,286 +1,201 @@
-import {ajax} from 'src/ajax';
-import adaptermanager from 'src/adaptermanager';
-import { config } from 'src/config';
-
-const bidmanager = require('src/bidmanager.js');
-const bidfactory = require('src/bidfactory.js');
-const CONSTANTS = require('src/constants.json');
-
-function RhythmoneAdapter (bidManager, global, loader) {
-  const version = '0.9.0.0';
-  let defaultZone = '1r';
-  let defaultPath = 'mvo';
-  let debug = false;
-  const placementCodes = {};
-  let loadStart;
-  let configuredPlacements = [];
-  const fat = /(^v|(\.0)+$)/gi;
-
-  if (typeof global === 'undefined') { global = window; }
-
-  if (typeof bidManager === 'undefined') { bidManager = bidmanager; }
-
-  if (typeof loader === 'undefined') { loader = ajax; }
-
-  function applyMacros(txt, values) {
-    return txt.replace(/\{([^\}]+)\}/g, function(match) {
-      var v = values[match.replace(/[\{\}]/g, '').toLowerCase()];
-      if (typeof v !== 'undefined') return v;
-      return match;
-    });
-  }
-
-  function load(bidParams, url, callback) {
-    loader(url, function(responseText, response) {
-      if (response.status === 200) {
-        callback(200, 'success', response.responseText);
-      } else {
-        callback(-1, 'http error ' + response.status, response.responseText);
-      }
-    }, false, {method: 'GET', withCredentials: true});
-  }
-
-  function flashInstalled() {
-    const n = global.navigator;
-    const p = n.plugins;
-    const m = n.mimeTypes;
-    const t = 'application/x-shockwave-flash';
-    const x = global.ActiveXObject;
-
-    if (p &&
-      p['Shockwave Flash'] &&
-      m &&
-      m[t] &&
-      m[t].enabledPlugin) { return true; }
-
-    if (x) {
-      try { if ((new global.ActiveXObject('ShockwaveFlash.ShockwaveFlash'))) return true; } catch (e) {}
-    }
-
-    return false;
-  }
-
-  var bidderCode = 'rhythmone';
+'use strict';
 
-  function attempt(valueFunction, defaultValue) {
-    try {
-      return valueFunction();
-    } catch (ex) {}
-    return defaultValue;
-  }
+import {registerBidder} from 'src/adapters/bidderFactory';
+import { BANNER, VIDEO } from 'src/mediaTypes';
 
-  function logToConsole(txt) {
-    if (debug) { console.log(txt); }
-  }
+function RhythmOneBidAdapter() {
+  this.code = 'rhythmone';
+  this.supportedMediaTypes = [VIDEO, BANNER];
 
-  function getBidParameters(bids) {
-    for (var i = 0; i < bids.length; i++) {
-      if (typeof bids[i].params === 'object' && bids[i].params.placementId) { return bids[i].params; }
-    }
-    return null;
-  }
+  this.isBidRequestValid = function (bid) {
+    return (bid.adUnitCode !== undefined);
```

Comment:
`adUnitCode` isn't available on the `bid` parameter here, so this is always failing, but `placementCode` is, if that's what you want to check for

Reply:
Are you sure?  I was seeing adUnitCode on the bid requests when this adapter was in the prebid 1.0 branch.  It's also mentioned in http://prebid.org/dev-docs/bidder-adapter-1.html

Reply:
Yeah, I was testing against master, on that branch the param is called `placementCode`. Looks like on 1.0 it is renamed `adUnitCode`. This would get merged into master so the safest thing might be to check for either one

Reply:
The adapter will use the first placementId that it finds on any of the bid requests if it's not configured individually.  The only requirement is for the configured bids to have at least one placement id.  The rest of the params are optional and exist for testing purposes only.

This is a side effect of our api being a single request architecture, and the url format only allows for one placement id.  We're currently working on a new api that addresses this limitation

Reply:
Got it.  Another question: what's the state of the master branch?  I was told via an email with some appnexus developers that I should check this 1.0 adapter into master.  Is master in a transitional state?

Reply:
Yes, both of those are correct, this adapter should be checked into the master branch, and master will soon be transitioned into the [1.0 branch](https://github.com/prebid/Prebid.js/tree/prebid-1.0). Any 1.0-compatible adapter in the (current) master branch will be transitioned/work in the 1.0 branch whenever that transition occurs, and RhythmOne would be included in that category after this PR is merged

Reply:
Cool, thank you